src/codemagic/models/certificate_p12_exporter.py

```diff
@@ -29,39 +27,42 @@
 
 
 class _OpenSsl:
+    __SUPPORTS_NOENC__: Optional[bool] = None
+
     def __init__(self):
         self._executable = shutil.which("openssl")
 
     def ensure_installed(self):
         if self._executable is None:
             raise IOError("OpenSSL executable is not present on system")
 
-    def get_version(self) -> Optional[Version]:
-        if not self._executable:
-            return None
+    def _is_noenc_flag_supported(self) -> bool:
+        """
+        OpenSSL version 3.0.0 deprecated option `-nodes` for disabling encryption
+        when invoking `openssl pkcs12`. It is replaced with `-noenc`.
+        See https://www.openssl.org/docs/man3.0/man1/openssl-pkcs12.html
 
-        try:
-            version_output = subprocess.check_output([self._executable, "version"])
-        except subprocess.CalledProcessError:
-            version_output = b""
+        Vanilla macOS (at least up to version 14.6) however still bundles `openssl`
+        executable that is using LibreSSL version that does not support `-noenc`.
 
-        version_match = re.search(r"\d+\.\d+\.\d+", version_output.decode(errors="ignore"))
-        if not version_match:
-            return None
+        Check the `-noenc` support by just parsing help message.
+        """
+
+        if self.__SUPPORTS_NOENC__ is None:
+            completed_process = subprocess.run(
+                (self._executable, "pkcs12", "-help"),
+                capture_output=True,
+                check=False,
+            )
+            self.__SUPPORTS_NOENC__ = b"-noenc" in completed_process.stdout or b"-noenc" in completed_process.stderr
```

Comment:
just out of curiosity, what are the cases when help text goes to stderr?

Reply:
Excellent question actually. LibreSSL doesn't support neither `-help` nor `help` and emits the usage to stderr:
```shell
% openssl version
LibreSSL 3.3.6
% openssl pkcs12 help
unknown option 'help'
usage: pkcs12 [-aes128 | -aes192 | -aes256 | -camellia128 |
    -camellia192 | -camellia256 | -des | -des3 | -idea]
    [-cacerts] [-CAfile file] [-caname name]
    [-CApath directory] [-certfile file] [-certpbe alg]
    [-chain] [-clcerts] [-CSP name] [-descert] [-export]
    [-in file] [-info] [-inkey file] [-keyex] [-keypbe alg]
    [-keysig] [-LMK] [-macalg alg] [-maciter] [-name name]
    [-nocerts] [-nodes] [-noiter] [-nokeys] [-nomac]
    [-nomaciter] [-nomacver] [-noout] [-out file]
    [-passin arg] [-passout arg] [-password arg] [-twopass]

 -aes128            Encrypt PEM output with CBC AES
 ...
 -nodes             Don't encrypt private keys
 ...
```

Perhaps this requires even a code comment?

Reply:
ha, that makes sense! and yes, commend in the code is totally helpful.

Reply:
but you search for `-noenc` in stderr just because you expect that LibreSSL will support it at some point but still without support of `-help`, is that correct? otherwise we can just check the first condition and immeddiately return the result.

Reply:
My hope (or assumtion?) is that maybe some day LibreSSL will follow OpenSSL and then both will support `-noenc`. As it is now, we'd get automatic support to the updated version.